src/main/java/org/commcare/suite/model/StackFrameStep.java

```diff
@@ -104,26 +105,27 @@ public String getValue() {
     public void addDataInstanceSource(ExternalDataInstanceSource source) {
         Objects.requireNonNull(source, String.format(
                 "Unable to add null instance data source to stack frame step '%s'", getId()));
-        String instanceId = source.getInstanceId();
-        if (dataInstanceSources.containsKey(instanceId)) {
+        String reference = source.getReference();
+        String referenceId = VirtualInstances.getReferenceId(reference);
+        if (dataInstanceSources.containsKey(referenceId)) {
```

Comment:
I think this is OK for now but in future we'll want to use the full ref in case there are 2 instances with the same refID:
```
jr://instance/remote/results
jr://instance/search-input/results
```